server/sonar-server/src/main/java/org/sonar/server/webhook/ws/CreateAction.java

```diff
@@ -0,0 +1,234 @@
+      .setName(name)
+      .setUrl(url)
+      .setCreatedAt(system.now());
+
+    if (project != null) {
+      checkNumberOfWebhook(numberOfWebhookOf(dbSession, project), "Maximum number of webhook reached for project '%s'", project.getKey());
+      dto.setProjectUuid(project.projectUuid());
+    } else {
+      checkNumberOfWebhook(numberOfWebhookOf(dbSession, organization), "Maximum number of webhook reached for organization '%s'", organization.getKey());
+      dto.setOrganizationUuid(organization.getUuid());
+    }
+
+    dbClient.webhookDao().insert(dbSession, dto);
+
+    return dto;
+  }
+
+  private static void writeResponse(Request request, Response response, WebhookDto element) {
+    Webhooks.CreateWsResponse.Builder responseBuilder = newBuilder();
+    responseBuilder.setWebhook(Webhook.newBuilder()
+      .setKey(element.getUuid())
+      .setName(element.getName())
+      .setUrl(element.getUrl()));
+
+    writeProtobuf(responseBuilder.build(), request, response);
+  }
+
+  private static void checkNumberOfWebhook(int nbOfWebhooks, String message, Object... messageArguments) {
+    if (nbOfWebhooks >= MAX_NUMBER_OF_WEBHOOKS){
+      throw new IllegalArgumentException(format(message, messageArguments));
+    }
+  }
+
+  private int numberOfWebhookOf(DbSession dbSession, OrganizationDto organization) {
+    return dbClient.webhookDao().selectByOrganizationUuid(dbSession, organization.getUuid()).size();
+  }
+
+  private int numberOfWebhookOf(DbSession dbSession, ComponentDto project) {
+    return dbClient.webhookDao().selectByProjectUuid(dbSession, project.uuid()).size();
+  }
+
+  private void checkUserPermissionOn(ComponentDto componentDto) {
+    userSession.checkComponentPermission(ADMIN, componentDto);
+  }
+
+  private void checkUserPermissionOn(OrganizationDto organizationDto) {
+    userSession.checkPermission(ADMINISTER, organizationDto);
+  }
+
+  private static void checkThatProjectBelongsToOrganization(ComponentDto componentDto, OrganizationDto organizationDto, String message, Object... messageArguments) {
+    if (!organizationDto.getUuid().equals(componentDto.getOrganizationUuid())) {
+      throw new NotFoundException(format(message, messageArguments));
+    }
+  }
+
+  private static void checkUrlPattern(String url, String message, Object... messageArguments) {
+    if (!url.toLowerCase(ENGLISH).startsWith("http://") && !url.toLowerCase(ENGLISH).startsWith("https://")) {
+      throw new IllegalArgumentException(format(message, messageArguments));
+    }
+    String sub = url.substring("http://".length());
+    if (sub.contains(":") && !sub.substring(sub.indexOf(':')).contains("@")) {
+      throw new IllegalArgumentException(format(message, messageArguments));
+    }
+  }
+
+  private OrganizationDto defaultOrganizationDto(DbSession dbSession) {
+    String uuid = defaultOrganizationProvider.get().getUuid();
+    return dbClient.organizationDao().selectByUuid(dbSession, uuid).get();
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Call "Optional#isPresent()" before accessing the value. ([squid:S3655](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3655))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2FMMF-647%2FSONAR-10345_manage_webhooks&issues=AWFv-apfh1v1MSKKr3nK&open=AWFv-apfh1v1MSKKr3nK)

[ISSUE_KEY]: AWFv-apfh1v1MSKKr3nK